Find a functionally equivalent code:android.graphics.BidiRenderer.setScriptFont:COMMENT
Method Modifier: private     static      
Comment:// TODO: Replace this method with one which returns the font based on the scriptCode.

@@ -1,7 +1,11 @@
 {
-    for (FontInfo fontInfo : fonts) {
-        if (fontInfo.mFont.canDisplayUpTo(text, run.start, run.limit) == -1) {
-            run.font = fontInfo;
+    for (Font font : fonts) {
+        if (font == null) {
+            logFontWarning();
+            continue;
+        }
+        if (font.canDisplayUpTo(text, run.start, run.limit) == -1) {
+            run.font = font;
             return;
         }
     }

